enclave_build/src/cache.rs

```diff
@@ -143,6 +146,102 @@ impl CacheManager {
         })
     }
 
+    /// Stores the image data provided as argument in the cache at the folder pointed
+    /// by the 'root_path' field.
+    pub fn store_image_data(&mut self, image_name: &str, image_data: &ImageData) -> Result<()> {
```

Comment:
This method is way too big, and complicated I would say? Can you think of some ways to simplify it, or maybe split it in multiple functions etc.?